src/tls/openssl/sni.c

```diff
@@ -109,14 +112,13 @@ struct tls_cert *tls_cert_for_sni(const struct tls *tls, const char *sni)
 
 		tls_cert = le->data;
 		x509 = tls_cert_x509(tls_cert);
-		if (!x509) {
-			tls_cert = NULL;
+		if (!x509)
 			continue;
-		}
 
 		nm = X509_get_subject_name(x509);
-		X509_NAME_get_text_by_NID(nm, NID_commonName, cn, (int) sz);
-		if (!str_cmp(sni, cn))
+		int n = X509_NAME_get_text_by_NID(nm, NID_commonName, cn,
+						  (int) sz);
+		if (n > 0 && !str_cmp(sni, cn))
```

Comment:
there is something strange here ... why is the buffer allocated to the same length as the input string ?

for example if the input sni is `space` and the common name is `spacex` will it match?

```
space
spacex
```

since the get_text_by_NID function returns number of chars in the string,
it is more efficient to use `str_ncmp` ...